src/components/Notification/__tests__/Notification.test.js

```diff
@@ -0,0 +1,359 @@
+    );
+
+    act(() => {
+      jest.runAllTimers();
+    });
+
+    expect(container.firstChild).toHaveStyleRule(
+      'background-color',
+      BG_COLOR_ERROR
+    );
+
+    rerender(
+      withAppContext(<Notification title="Foo bar" variant={VARIANT_SUCCESS} />)
+    );
+
+    act(() => {
+      jest.runAllTimers();
+    });
+
+    expect(container.firstChild).toHaveStyleRule(
+      'background-color',
+      BG_COLOR_SUCCESS
+    );
+  });
+
+  it('does not use timeouts to time navigation actions for VARIANT_ERROR', () => {
+    expect(getSetTimeoutCalls()).toHaveLength(0);
+
+    render(
+      withAppContext(<Notification title="Foo bar" variant={VARIANT_ERROR} />)
+    );
+
+    act(() => {
+      jest.runAllTimers();
+    });
+
+    expect(getSetTimeoutCalls()).toHaveLength(0);
+  });
+
+  it('uses timeouts to time navigation actions for VARIANT_NOTICE', () => {
+    expect(getSetTimeoutCalls()).toHaveLength(0);
+
+    render(
+      withAppContext(<Notification title="Foo bar" variant={VARIANT_NOTICE} />)
+    );
+
+    act(() => {
+      jest.runAllTimers();
+    });
+
+    expect(getSetTimeoutCalls()).toHaveLength(2);
+  });
+
+  it('uses timeouts to time navigation actions for VARIANT_SUCCESS', () => {
+    expect(getSetTimeoutCalls()).toHaveLength(0);
+
+    render(
+      withAppContext(<Notification title="Foo bar" variant={VARIANT_SUCCESS} />)
+    );
+
+    act(() => {
+      jest.runAllTimers();
+    });
+
+    expect(getSetTimeoutCalls()).toHaveLength(2);
+  });
+
+  it('does not use timeouts to time navigation actions for TYPE_GLOBAL', () => {
+    expect(getSetTimeoutCalls()).toHaveLength(0);
+
+    render(
+      withAppContext(<Notification title="Foo bar" type={TYPE_GLOBAL} />)
+    );
+
+    act(() => {
+      jest.runAllTimers();
+    });
+
+    expect(getSetTimeoutCalls()).toHaveLength(0);
+  });
+
+  it('hides the component after a specific amount of time and executes callback function', () => {
+    expect(getSetTimeoutCalls()).toHaveLength(0);
+    expect(getClearTimeoutCalls()).toHaveLength(0);
+
+    const onClose = jest.fn();
+
+    const { container } = render(
+      withAppContext(<Notification title="Foo bar" onClose={onClose} />)
+    );
+
+    expect(getSetTimeoutCalls()).toHaveLength(2);
+    expect(getClearTimeoutCalls()).toHaveLength(0);
+
+    expect(onClose).not.toHaveBeenCalled();
+
+    expect(container.firstChild.classList.contains('slideup')).toEqual(false);
+
+    act(() => {
+      jest.advanceTimersByTime(ONCLOSE_TIMEOUT);
+    });
+
+    expect(onClose).not.toHaveBeenCalled();
+
+    act(() => {
+      jest.advanceTimersByTime(SLIDEUP_TIMEOUT);
+    });
+
+    expect(onClose).toHaveBeenCalled();
+
+    expect(getSetTimeoutCalls()).toHaveLength(2);
+    expect(getClearTimeoutCalls()).toHaveLength(2);
+
+    expect(container.firstChild.classList.contains('slideup')).toEqual(true);
+  });
+
+  it('clears timeouts on unmount', () => {
+    expect(getSetTimeoutCalls()).toHaveLength(0);
```

Comment:
Could we move this to after the `act(runAllTimers)` ?

Reply:
I'll remove the check for `getSetTimeoutCalls`,  because I don't need to be checking for that. Just for `clearTimeout` calls.